index.d.ts

```diff
@@ -1668,6 +1672,54 @@ declare namespace Moleculer {
 		}
 	}
 
+	/**
+	 * you can use multiple modifier with a dot
+	 * e.g. `black.bgRed.bold`
+	 */
+	type KleurColor = keyof Kleur | string;
+
+	type ActionLoggerOptions = {
+		logger?: LoggerInstance;
+		logLevel?: LogLevels;
+		logParams?: boolean;
+		logResponse?: boolean;
+		logMeta?: boolean;
+		folder?: string | null;
+		extension?: string;
+		colors?: {
+			request?: KleurColor;
+			response?: KleurColor;
+			error?: KleurColor;
+		};
+		whitelist?: Array<string>;
+	};
+	type TransitLoggerOptions = {
+		logger?: LoggerInstance;
+		logLevel?: LogLevels;
+		logPacketData?: boolean;
+		folder?: string | null;
+		extension?: string;
+		colors?: {
+			receive?: KleurColor;
+			send?: KleurColor;
+		};
+		packetFilter?: Array<Packets.packetType>;
+	};
+	type CompressionOptions = {
+		method?: "deflate" | "deflateRaw" | "gzip";
+		threshold?: number | string;
+	};
+	export const Middlewares: {
+		Debugging: {
+			ActionLogger(options?: ActionLoggerOptions): Middleware;
+			TransitLogger(options?: TransitLoggerOptions): Middleware;
+		};
+		Transmit: {
+			Compression(options?: CompressionOptions): Middleware;
+			Encryption(password: string, algorithm: string, iv: string | Buffer): Middleware;
```

Comment:
Looks like algorithm is used to call into `crypto.createCipheriv` and `crypto.createCipher` which accepts a type of `crypto.CipherCCMTypes`.

Reply:
`iv` is of type `crypto.BinaryLike`.

Reply:
there is even more complexe, it can be `crypto.CipherCCMTypes` / `crypto.CipherOCBTypes` / `crypto.CipherGCMTypes`, also the password length will depend on the algorithm . 

And more, this middleware is a bad one . The encryption without IV is deprecated (not really secure), and reusing the same IV for all messages is also not secure . ( IV need to be included in the message, and change on every messages https://medium.com/swlh/an-introduction-to-the-advanced-encryption-standard-aes-d7b72cc8de97#:~:text=Initialization%20vector%20(IV),is%20an%20added%20security%20layer. ) .